jenkins_jobs/local_yaml.py

```diff
@@ -597,11 +597,25 @@ def format(self, **kwargs):
         self._template.environment.loader = self._loader
         return self._template.render(kwargs)
 
+    def get_object_to_format(self):
+        return self
+
+
+class LateYamlLoader(CustomLoader):
+    """A loader for data rendered via Jinja2, to be loaded as YAML and then deep formatted."""
+
+    def __init__(self, yaml_str, loader):
+        self._yaml_str = yaml_str
+        self._loader = loader
+
+    def get_object_to_format(self):
+        return load(self._yaml_str, search_path=self._loader._search_path)
+
 
 class Jinja2YamlLoader(Jinja2Loader):
```

Comment:
Not sure if we discussed before, but would it be possible to join j2 and j2-yaml?

Reply:
The maintainers of JJB mentioned this in my upstream change but it's not easy because it might break some existing jobs.